juju/juju
# chore: remove unused methods from instancepoller facade

The following facade methods on the `instancepoller` facade are currently unused:
- `ProviderAddresses`, `SetProviderAddresses`
- `ModelConfig`, `WatchForModelConfigChanges` (inherited from `common.MongoModelWatcher`)

This PR removes these unused methods from the facade. I am told that we don't need to bump the facade version, since this is a controller facade.

This allows us to remove a bunch of unnecessary dependencies from `InstancePollerAPI`. In particular, this facade no longer needs to access model config.

Drive-bys:
- Move existing service interface definitions (`ControllerConfigService`, `NetworkService`) into a new file `service.go`, and put the generated mocks into `service_mock_test.go`, as per the new convention.
- Change the instancepoller tests to use generated service mocks, and delete unneeded tests for deleted methods.

## Checklist




## QA steps



Check that model migration still works.
```
juju bootstrap lxd c2
juju bootstrap lxd c1
juju add-model m
juju deploy tiny-bash
juju migrate m c2
```

## Links



**Jira card:** JUJU-6102
